src/main/java/org/reactome/addlinks/dataretrieval/ensembl/EnsemblBiomartRetriever.java

```diff
@@ -0,0 +1,232 @@
+        final int MAX_QUERY_RETRIES = 5;
+        // Create connection to Biomart URL for each species, retrieving a list of microarray probe types, if available.
+        URL biomartUrlWithSpecies = new URL(queryString);
+        HttpURLConnection biomartConnection = (HttpURLConnection) biomartUrlWithSpecies.openConnection();
+        if (biomartConnection.getResponseCode() == HttpStatus.SC_OK) {
+            Set<String> biomartResponseLines = new HashSet<>();
+            BufferedReader br = new BufferedReader(new InputStreamReader(biomartConnection.getInputStream()));
+            String line;
+            while ((line = br.readLine()) != null) {
+                // Biomart still responds with a 200, even if no data exists. For now, we handle it by
+                // checking the returned content for the string 'ERROR'. It will retry up to 5 times, with a 10 second delay.
+                if (line.contains("ERROR") ) {
+                    retryCount++;
+                    if (retryCount < MAX_QUERY_RETRIES) {
+                        return retryQuery(queryString, retryCount);
+                    }
+                    // The data does not exist in BioMart for a few species. Frog (X. tropicalis) has UniProt data but not Microarray data;
+                    // Yeast (S. cerevisiae) has Uniprot-SwissProt data but not UniProt-TrEMBL data; Yeast (S. pombe), P. falciparum
+                    // and D. discoideum don't have UniProt or Microarray data, all at time of writing (January 2020).
+//                    throw new Exception("Biomart query failed with message: " + line + "\nThis can happen without issue for certain species (X. tropicalis, S. pombe, S. cerevisiae, P. falciparum) because the data doesn't exist in BioMart");
+                    throw new BioMartQueryException(line +
+                            "\nThis can happen without issue for certain species (X. tropicalis, S. pombe, S. cerevisiae, P. falciparum) " +
+                            "because the data doesn't exist in BioMart");
+                } else {
+                    biomartResponseLines.add(line);
+                }
+            }
+            return biomartResponseLines;
+
+        } else {
+            retryCount++;
+            if (retryCount < MAX_QUERY_RETRIES) {
+                return retryQuery(queryString, retryCount);
+            }
+            throw new HttpException(
+                    "Unable to connect to Biomart (" +
+                            biomartConnection.getResponseCode() + ": " + biomartConnection.getResponseMessage() +
+                            ") with URL: " + queryString
+            );
+        }
+    }
+
+    // Recursive method that retries the BioMart query.
+    private Set<String> retryQuery(String queryString, int retryCount) throws InterruptedException, IOException, HttpException, BioMartQueryException {
+        final long QUERY_SLEEP_DURATION = Duration.ofSeconds(5).toMillis();
```

Comment:
I noticed the comment above says the sleep time will be for 10 seconds, but it is given as 5 here.